channeldb/channel.go

```diff
@@ -589,36 +589,30 @@ func (c *OpenChannel) hasChanStatus(status ChannelStatus) bool {
 }
 
 // RefreshShortChanID updates the in-memory short channel ID using the latest
-// value observed on disk.
+// value observed on disk. Since a new short channel ID is a result of a
+// funding tx confirmation, we also update the IsPending value that might have
+// been changed as well.
 func (c *OpenChannel) RefreshShortChanID() error {
 	c.Lock()
 	defer c.Unlock()
 
-	var sid lnwire.ShortChannelID
 	err := c.Db.View(func(tx *bbolt.Tx) error {
 		chanBucket, err := fetchChanBucket(
 			tx, c.IdentityPub, &c.FundingOutpoint, c.ChainHash,
 		)
 		if err != nil {
 			return err
 		}
-
-		channel, err := fetchOpenChannel(chanBucket, &c.FundingOutpoint)
-		if err != nil {
-			return err
+		if err := fetchChanInfo(chanBucket, c); err != nil {
```

Comment:
Althought I have no familiarity with this part of the code yet, I noticed that this change is way broader than what the function name implies. Users of `RefreshShortChanID` function would expect that it only refreshes `c.ShortChannelID` and `c.IsPending` as documented, yet many other members are changed as well.

Reply:
@bhandras  You are right, I was asked by the lightning lab team not to change the function name as major refactoring is scheduled for this part.

Reply:
Could you please add a TODO to the doc so that it's clear that this is intentional?

Reply:
>  I was asked by the lightning lab team not to change the function name as major refactoring is scheduled for this part.

The prior diff for this PR was more minimal: it only also refreshed the short channel ID and pending state of the channel. Why the departure?

Reply:
> The prior diff for this PR was more minimal: it only also refreshed the short channel ID and pending state of the channel. Why the departure?

Although not minimal as the prior diff I think it is a more correct approach then selecting specific fields.
I also think it is not risky as this exact loading logic is executed on startup anyway and it seems ok to consider the db as the single source of truth regarding the channel state.

Reply:
> Could you please add a TODO to the doc so that it's clear that this is intentional?

Added.

Reply:
I'm fine with this after adding the comment. Not entirely sure if the refactor is affected though.